Resources improvements

MeuhV1 opened the pull request:
- Hoovering over culture bar shows culture rank
- Fixed bug that caused buildings that provide health/happiness from resources to only work when the culture rank allowed it
- CvCity::getBonusHappiness() and CvCity::getBonusHealth() take culture rank into account.

Improved the city resource display:
- Insufficient resources that provide health/happiness with buildings have a health/happiness icon

Updated BUG Domestic resource advisor
- Included new DoC resources (like tobacco)
- Ordered resources by improvement
- Fixed Hit Singles, Hit Movies, Hit Football
- Resources page distinguishes between not available (red -), hasBonus (yellow o) and hasBonusEffect (green x)

dguenms commented on CvGameCoreDLL/CvCity.cpp:
```diff
@@ -4260,6 +4262,31 @@ void CvCity::processBonusEffect(BonusTypes eBonus, int iChange)
 
 
+// Leoreth
+void CvCity::processBonusEffect(BonusTypes eBonus, int iChange)
```
Looks like the diff is messed up here, can you explain in more detail what changes you made? Preferably by annotating the corresponding lines.

dguenms reviewed (changes requested):
That's a really cool set of changes, I already expected that there were many interface elements that I did not consider changing.

MeuhV1 commented:
The diff is indeed messy. I basically moved the parts between ** (both for loops) from procesBonusEffect() to procesBonus(). (And added the required int declarations to make it work) That makes sure that the buildings that provide health/happiness with resources only take into account if the city has access to a bonus, instead of having also the required culture rank.

```
// Leoreth
void CvCity::processBonusEffect(BonusTypes eBonus, int iChange)
{
	int iI;
	int iValue;
	int iGoodValue;
	int iBadValue;

	iValue = GC.getBonusInfo(eBonus).getHealth();
	iGoodValue = std::max(0, iValue);
	iBadValue = std::min(0, iValue);

	**for (iI = 0; iI < GC.getNumBuildingInfos(); iI++)
	{
		iValue = GC.getBuildingInfo((BuildingTypes) iI).getBonusHealthChanges(eBonus) * getNumActiveBuilding((BuildingTypes)iI);

		if (iValue >= 0)
		{
			iGoodValue += iValue;
		}
		else
		{
			iBadValue += iValue;
		}
	}**

	changeBonusGoodHealth(iGoodValue * iChange);
	changeBonusBadHealth(iBadValue * iChange);


	iValue = GC.getBonusInfo(eBonus).getHappiness();
	iGoodValue = std::max(0, iValue);
	iBadValue = std::min(0, iValue);

	**for (iI = 0; iI < GC.getNumBuildingInfos(); iI++)
	{
		iValue = getNumActiveBuilding((BuildingTypes)iI) * GC.getBuildingInfo((BuildingTypes) iI).getBonusHappinessChanges(eBonus);

		if (iValue >= 0)
		{
			iGoodValue += iValue;
		}
		else
		{
			iBadValue += iValue;
		}
	}**

	changeBonusGoodHappiness(iGoodValue * iChange);
	changeBonusBadHappiness(iBadValue * iChange);
}
```

dguenms commented:
Okay, thanks for the clarification. I assumed that's what you did, just wanted to make sure I didn't miss any other changes.

Unfortunately I rewrote the history of the entire branch in the meantime, including my own code for the red coloured -1 in the interface screen that I hadn't pushed yet by the time of your pull request. That's why all my commits show up in this PR again.

I think the easiest way to make this branch compatible again is to break up your commit again (keeping the changes themselves) with `git reset HEAD^` and then copying the diff with `git stash`. Afterwards you can pull the current resources branch again and paste the copied diff with `git stash pop`. You may have a conflict in the DLL binary but that can be easily solved by compiling again.

Commit:
- Hoovering over culture bar shows culture rank
- Fixed bug that caused buildings that provide health/happiness from resources to only work when the culture rank allowed it
- CvCity::getBonusHappiness() and CvCity::getBonusHealth() take culture rank into account.
- Insufficient resources that provide health/happiness with buildings have a health/happiness icon

Updated BUG Domestic resource advisor:
- Included new DoC resources (like tobacco)
- Ordered resources by improvement
- Fixed Hit Singles, Hit Movies, Hit Football
- Resources page distinguishes between not available (red -), hasBonus (yellow o) and hasBonusEffect (green x)

MeuhV1 commented:
Updated.
